contracts/src/Teleporter/upgrades/TeleporterRegistry.sol

```diff
@@ -137,37 +139,47 @@ contract TeleporterRegistry {
      * @dev Gets the address of a protocol version.
      * Requirements:
      *
-     * - `version` must be a valid version, i.e. greater than 0 and not greater than the latest version.
+     * - `version` must be a valid version registered.
      */
     function getAddressFromVersion(
         uint256 version
     ) public view returns (address) {
         require(version != 0, "TeleporterRegistry: zero version");
+        address protocolAddress = _versionToAddress[version];
         require(
-            version <= latestVersion,
-            "TeleporterRegistry: invalid version"
+            protocolAddress != address(0),
+            "TeleporterRegistry: version not found"
         );
-        return _versionToAddress[version];
+        return protocolAddress;
     }
 
     /**
      * @dev Gets the version of the given `protocolAddress`.
-     * If `protocolAddress` is not a registered protocol address, returns 0, which is an invalid version.
+     * Rrequirements:
+     *
+     * - `protocolAddress` must be a valid protocol address registered.
      */
     function getVersionFromAddress(
         address protocolAddress
     ) public view returns (uint256) {
-        return _addressToVersion[protocolAddress];
+        require(
+            protocolAddress != address(0),
+            "TeleporterRegistry: zero protocol address"
+        );
+        uint256 version = _addressToVersion[protocolAddress];
+        require(version != 0, "TeleporterRegistry: protocol address not found");
+        return version;
     }
 
     /**
      * @dev Adds the new protocol version address to the registry.
      * Updates latest version if the version is greater than the current latest version.
      *
      * Emits a {AddProtocolVersion} event when successful.
+     * Emits a {LatestVersionUpdated} event when a new protocol version greater than the current latest version is added.
      * Note: `protocolAddress` doesn't have to be a contract address, this is primarily
```

Comment:
I'm not sure this note is necessary to include here. I think what it's describing is that the contract to be registered does not have to actually be deployed. But the registry doesn't call any of the registered contracts directly, so I don't think the language in the registry contract needs to make the distinction between deployed and undeployed contracts at all.

If you feel like this comment should stay, I'd suggest simplifying it to something like "entry.protocolAddress does not need to be a deployed contract. This allows the registry to be updated ahead of a protocol upgrade being deployed"

Reply:
I wanted to leave the note about this being possibly used in case where its a patch for the current version, updated the wording.